lib/info-extras.js

```diff
@@ -124,11 +124,49 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', ',\n');
+  let watchNextJson;
   try {
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let rvsParams = jsonStr.rvs.split(',').map((e) => {
+    return {
+      id: util.between(e, 'id=', '&'),
+      shortViewCount: util.between(e, 'short_view_count_text=', '+views')
+    };
+  })
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    let details;
+    if (details = result.compactVideoRenderer) {
+      try {
+        let viewCount = details.viewCountText.simpleText;
+        if(!viewCount)
+          viewCount = details.viewCountText.runs[0].text.replace(' watching', '');
+        let shortViewCount = details.shortViewCountText.simpleText;
+        if(shortViewCount === 'Recommended for you')
```

Comment:
using `rvs` would also be more reliable than checking for strings like these, since these are language dependent

Reply:
Thank you for your review. I will check that out.

Reply:
I think the rvs_params contains only the recommended videos not related videos @fent

Reply:
![image](https://user-images.githubusercontent.com/10515204/64090509-1f2fd780-cd69-11e9-8321-0f9747b1a5f5.png)

These data are obtained from the rvs_params.

Reply:
ah, so it shows up in some weird format sometimes? for me it shows this

<img width="561" alt="Screen Shot 2019-09-02 at 11 04 10 AM" src="https://user-images.githubusercontent.com/933490/64123118-6f129c80-cd71-11e9-9d26-cf2a6a3513d6.png">

anyway, if we use this data instead of `rvsParams`, we can check if the string contains a number at the start, with like `/^\d/` and then we can assume it's not "Recommended for you" if it matches.

> I think the rvs_params contains only the recommended videos not related videos

aren't those the same?

Reply:
> aren't those the same?
From my observation, only a part of the ones obtained from `compactVideoRenderer` is available in the `rvs_params` only the ones with `shortViewCount` as `Recommended For You`

Reply:
> anyway, if we use this data instead of `rvsParams`, we can check if the string contains a number at the start, with like `/^\d/` and then we can assume it's not "Recommended for you" if it matches.

I will update that